zebra-chain/src/transparent/script.rs

```diff
@@ -12,7 +12,20 @@ use std::{fmt, io};
     any(test, feature = "proptest-impl"),
     derive(proptest_derive::Arbitrary)
 )]
-pub struct Script(pub Vec<u8>);
+pub struct Script(Vec<u8>);
+
+impl Script {
+    /// Create a new Bitcoin script from its raw bytes.
+    /// The raw bytes must not contain the compact size prefix.
+    pub fn new(raw_bytes: Vec<u8>) -> Self {
+        Script(raw_bytes)
+    }
+
+    /// Return the raw bytes of the script without the compact size prefix.
+    pub fn raw_bytes(&self) -> &Vec<u8> {
```

Comment:
Similarly, can we restrict this function to a smaller scope?
```suggestion
    fn raw_bytes(&self) -> &Vec<u8> {
```

Reply:
Changed to return a slice.

Unfortunately it's not possible to restrict it since it is used by `zebra_script`.